lib/dmv_factory.rb

```diff
@@ -0,0 +1,24 @@
+class DmvFactory 
+  attr_reader :office_list
+  def initialize
+    @office = nil
+  end
+
+  def create_offices_helper(api_data)
+    @office_list = api_data.map do |office|
+     office_data = { 
+      name: "#{office[:dmv_id]} #{office[:dmv_office]}", 
+      address: "#{office[:address_li]} #{office[:address__1]} 
+                #{office[:city]} #{office[:state]} #{office[:zip]}", 
+      phone: office[:phone], services: office[:services_p] } 
+    end
+  end
+
+
+  def create_offices(api_data)
+    self.create_offices_helper(api_data)
+    offices = @office_list.map do |office|
+      facility = Facility.new(office)
+    end
+  end
```

Comment:
I don't think you need both the `create_offices_helper` and `create_offices` method. They could both be combined. This would make the code a lot easier to read.

```ruby
def create_offices_helper(api_data)
  api_data.map do |office|
    office_data = { 
      name: "#{office[:dmv_id]} #{office[:dmv_office]}", 
      address: "#{office[:address_li]} #{office[:address__1]} #{office[:city]} #{office[:state]} #{office[:zip]}", 
      phone: office[:phone], 
      services: office[:services_p] 
    } 

    Facility.new(office_data)
  end
end
```

You also don't need to store the result of the enumerable into a variable like you are doing on line 8 and line 20.